app/src/main/java/com/kickstarter/ui/activities/EditorialActivity.kt

```diff
@@ -1,71 +1,84 @@
 package com.kickstarter.ui.activities
 
 import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.isGone
 import com.jakewharton.rxbinding.view.RxView
 import com.kickstarter.R
 import com.kickstarter.databinding.ActivityEditorialBinding
-import com.kickstarter.libs.BaseActivity
-import com.kickstarter.libs.qualifiers.RequiresActivityViewModel
-import com.kickstarter.libs.rx.transformers.Transformers.observeForUI
+import com.kickstarter.libs.rx.transformers.Transformers.observeForUIV2
+import com.kickstarter.libs.utils.extensions.addToDisposable
+import com.kickstarter.libs.utils.extensions.getEnvironment
 import com.kickstarter.ui.fragments.DiscoveryFragment
 import com.kickstarter.viewmodels.EditorialViewModel
+import io.reactivex.disposables.CompositeDisposable
 
-@RequiresActivityViewModel(EditorialViewModel.ViewModel::class)
-class EditorialActivity : BaseActivity<EditorialViewModel.ViewModel>() {
+class EditorialActivity : AppCompatActivity() {
+
+    private lateinit var factory: EditorialViewModel.Factory
+    private val viewModel: EditorialViewModel.EditorialViewModel by viewModels { factory }
+    private val disposables = CompositeDisposable()
 
     private lateinit var binding: ActivityEditorialBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityEditorialBinding.inflate(layoutInflater)
 
+        getEnvironment()?.let { env ->
+            intent?.let {
+                factory = EditorialViewModel.Factory(env, it)
+            }
+        }
+
```

Comment:
Suggestion, not mandatory, for Activities we could add the networking check and the back navigation:

`setUpConnectivityStatusCheck(lifecycle)`
`this.onBackPressedDispatcher.addCallback {
            finishWithAnimation()
        }` ---> mainly to have a consistant back navigation experience